interface/src/components/homepage.js

```diff
@@ -1,8 +1,35 @@
 import React, { Component } from 'react';
-import { Link } from 'react-router-dom'
+import { withRouter } from 'react-router-dom'
+import { getUserData } from './requests'
 import './stylesheets/homepage.css'
 
-class homepage extends Component{
+class Homepage extends Component{
+    constructor(props){
+        super(props);
+        this.updateUser=this.updateUser.bind(this);
+    }
+
+    async updateUser(){
+        let apikey= document.getElementById("api-key").value;
+        if(apikey !== ""){
+            let user= await getUserData(apikey);
+            if(user.hasOwnProperty("error")){
+                return alert('Encountered error: '+ user.error);
+            }
+            if(Object.keys(user).length === 0)
+                return alert('User not found. Please enter a valid API key');
+            let isValid = window.confirm(`Please confirm, are you ${user}?`);
+            if(isValid === true){
+                this.props.setUser(user);
+                return this.props.history.push("/upload1");
+            }
+            return;
+        }
+        if(this.props.user !== undefined)
+        return this.props.history.push("/upload1");
+        return alert('Please enter a valid API key');
```

Comment:
IMO it would be better to have an Error Message component display the error, rather than an alert. This way focus can also be set to the control that needs better input and/or other UI decorations could also be set to highlight whats wrong

Reply:
Actually, there is a component for errors ([interface/src/components/error.js](https://github.com/PecanProject/BETYdb-YABA/blob/688fcfe3f7e08535013586c06d0689ccb0f4d114/interface/src/components/error.js)) but we just want to overlay a message for an invalid API key rather redirecting to an error page